pkg/server/app/app.go

```diff
@@ -46,6 +47,7 @@ type K8sOperations interface {
 	SetNamespaceAnnotations(namespace string, annotations map[string]string) error
 	DeleteDeployEnvVars(namespace, name string, evNames []string) error
 	CreateOrUpdateDeployEnvVars(namespace, name string, evs []*EnvVar) error
+	FindAppByLabel(label string) ([]string, error)
```

Comment:
Do you think this is really necessary? Because if we are looking for the name from app, why use Namespace in func? Although in your case the app name is the same the namespace.